### PR TITLE
[bugfix] model trainer device conflict

### PR DESCRIPTION
**Pull request by jrzaurin**

### CODE REVIEW

File: pytorch_widedeep/models/wide_deep.py [hunk 1]
```diff
@@ -154,6 +151,10 @@ def __init__(
             with_fds,
         )
 
+        # this attribute will be eventually over-written by the Trainer's
+        # device. Acts here as a 'placeholder'.
+        self.wd_device = "cpu"
```
Comment (gradientsky) [inline pytorch_widedeep/models/wide_deep.py:156]: Thinking about error-proofing: if we do expect this value to be set by the caller, it's probably not a good idea to use a default here - it can mask a defect if the setting never happened. Quick fail might be a preferred approach since we immediately know about it in unit tests. My suggestion is setting this to `None` here, and then adding a precondition check that the device is set in the trainer's `fit` call.

**Review by gradientsky — APPROVED** [review]

**Review by gradientsky — APPROVED** [review]

File: pytorch_widedeep/models/wide_deep.py (hunk 1 above)
Comment (jrzaurin) [reply pytorch_widedeep/models/wide_deep.py]: It is now set to None. 

Regarding to pre-conditioning set, in reality setting up the `wd_device` attribute happens manually (i.e. is enforced) when instantiating the `Trainer` class, here: https://github.com/jrzaurin/pytorch-widedeep/blob/a80da76c721a4c22192433d10b03828d6b981d6b/pytorch_widedeep/training/trainer.py#L241

Therefore, it is impossible that `fit` is called without `wd_device` set. Although this set up might look (perhaps) counter-intuitive is, actually convenient for an upcoming `BaseTrainer` class, with all the checks and model set ups will happen in a private method in base class.
